internal/cephfs/store/volumeoptions.go

```diff
@@ -236,6 +236,16 @@ func NewVolumeOptions(
 	opts.SubvolumeGroup = clusterData.CephFS.SubvolumeGroup
 	opts.Owner = k8s.GetOwner(volOptions)
 
+	// backingSnapshot will be used only with read-only access modes
+	volCaps := req.GetVolumeCapabilities()
```

Comment:
Could you make this a helper function so that testing is easier?

```go
    // default BackingSnapshot to true only for ReadOnly volume requests
    opt.BackingSnapshot = isVolumeCreateRO(req)
    ...
}

func isVolumeCreateRO(req *csi.CreateVolumeRequest) bool {
    ...
}
```

That way, you can add a unit test for `isVolumeCreateRO()`, document it's behaviour and reduce the complexity of `NewVolumeOptions()`.